### ISSUE TITLE
6 hour doc review notes

### ISSUE THREAD

**Issue by georgescharlesbrain** [ChainMovers/suibase#66]
I reviewed the docs from start to finish and took notes using hypothesis whenever something came up.

You can find my notes via [https://hypothes.is/search?q=suibase.io](https://hypothes.is/search?q=suibase.io).

I hope it can be of any use.

Thx for the toolset, much appreciated.

**Comment by georgescharlesbrain** [comment]
Can be closed for me if the link is not required anymore.

**Comment by mario4tier** [comment]
Please leave the hypothesis link up for a while, and will cut&paste here as I follow up on individual items.

Might take a few weeks to go through it all (have a few big ones to address before).

**Comment by georgescharlesbrain** [comment]
The hypothesis comments are public. Anyone who goes to the website and has the extension active will see the comments, I think.

**Comment by mario4tier** [comment]
Some answers

**| Why the config-default and sui-repo-default folders?**

We are getting into some implementation details.

Suibase have a "default behavior" of updating/running the latest code from Mysten Labs. This repo is always (and the only one) that can be stored in sui-repo-default.

The Suibase convention specify the repo should be at ```~/suibase/workdirs/sui-repo```, and by default this is implemented as a soft link (redirection) to ```sui-repo-default```.

For some advanced user, they might not want that "default behavior" and prefer to have a 100% control over all the git operations (e.g. because doing modification to the Sui code).

In that case they would want the soft link to point to their repo instead of "sui-repo-default".

From a user perspective, controlling that soft-link is done by the command "set-sui-repo" (e.g. localnet set-sui-repo --path my_own_repo).

The same is done for the "config" soft-link which is redirecting by default to "config-default". This is to not mix a user controlled sui.keystore with the default one.

"set-sui-repo" is a feature used in particular by fastfrank who has to constantly switch between multiple branches for its localnet.


**| Why does local dev and test have different addresses? I can see the | benefit of having different address on the mainnet. But doesnt it get more complex if each net has its own set of addresses?**

Main intent is to have the localnet use the same addresses for everyone. Just a simplification for quickly hardcoding addresses when sharing code etc... and expect that it will just work on anyone setup with a cut&paste.

That main intent would be incompatible with your proposal since dev/testnet addresses need to remain unique to each user.

May be we could have applied your proposal to dev/testnet only (and make them same but unique to each user)... but I went with a different solution. Instead of **same address**, I added a "name service" layer allowing for **same name**... across all networks.

So, an app can get the 15 generated addresses "by name" using the Suibase Helper API. This is IMHO way better when writing large test plans that runs in parallel (instead of hardcoding various address).

More info: https://chainmovers.github.io/suibase-api-docs/suibase/struct.Helper.html#method.client_sui_address

That "name service" feature may evolve one day for the user to attach name to its own private address.

**| What is the advice for your original sui binary? Should this be removed? cons and benefits? How much space is this taking up? The sui binary takes up 100MB**

Suibase won't touch the already installed binary and sui.keystore. I just design it such that they can be side-by-side.

Up to the user to decide what to do with other Sui installation outside Suibase.
